src/NServiceBus.Core/Transports/Msmq/Msmq.cs

```diff
@@ -64,22 +66,39 @@ ReceiveStrategy SelectReceiveStrategy(TransactionSupport minimumConsistencyGuara
         }
 
         /// <summary>
+        /// Performs any required start-up checks for receiving.
+        /// </summary>
+        protected internal override Task PrepareForReceiving(PrepareForReceivingContext context)
+        {
+            return new QueuePermissionChecker().CheckQueuePermissions(context.ReceiveQueues);
+        }
+
+        /// <summary>
+        /// Performs any required start-up checks for sending..
+        /// </summary>
+        protected internal override Task PrepareForSending(PrepareForSendingContext context)
+        {
+            new TimeToBeReceivedOverrideCheck(context.Settings).CheckTimeToBeReceivedOverrides();
+            return new QueuePermissionChecker().CheckQueuePermissions(context.SendQueues);
+        }
+
+        /// <summary>
         /// Configures transport for sending.
         /// </summary>
-        protected internal override void ConfigureForSending(TransportSendingConfigurationContext context)
+        protected internal override TransportSendingConfigurationResult ConfigureForSending(TransportSendingConfigurationContext context)
```

Comment:
Why not just change it to?

`TransportSendingConfigurationContext ConfigureForSending`